modules/auxiliary/admin/kerberos/keytab.rb

```diff
@@ -0,0 +1,201 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Kerberos keytab utilities',
+        'Description' => %q{
+          Utilities for interacting with keytab files, which can store the hashed passwords of one or
+          more principals.
+
+          Discovered keytab files can be used to generate Kerberos Ticket Granting Tickets, or bruteforced
+          offline.
+
+          Keytab files can be also useful for decrypting Kerberos traffic using Wireshark dissectors,
+          including the krbtgt encrypted blobs if the AES password hash is used.
+        },
+        'Author' => [
+          'alanfoster' # Metasploit Module
+        ],
+        'References' => [
+        ],
+        'License' => MSF_LICENSE,
+        'Notes' => {
+          'Stability' => [],
+          'SideEffects' => [],
+          'Reliability' => [],
+          'AKA' => []
+        },
+        'Actions' => [
+          ['LIST', { 'Description' => 'List the entries in the keytab file' }],
+          ['ADD', { 'Description' => 'Add a new entry to the keytab file' }],
+        ],
+        'DefaultAction' => 'LIST',
+        'DefaultOptions' => {
+          'VERBOSE' => true
+        }
+      )
+    )
+
+    supported_encryption_names = ['ALL']
+    supported_encryption_names += Rex::Proto::Kerberos::Crypto::Encryption::SUPPORTED_ENCRYPTIONS
+                                  .map { |id| Rex::Proto::Kerberos::Crypto::Encryption.const_name(id) }
+
+    register_options(
+      [
+        OptString.new('KEYTAB_FILE', [true, 'The keytab file to manipulate']),
+        OptString.new('PRINCIPAL', [false, 'The kerberos principal name']),
+        OptString.new('REALM', [false, 'The kerberos realm']),
+        OptEnum.new('ENCTYPE', [false, 'The enctype to use. If a password is specified this can set to \'ALL\'', supported_encryption_names[0], supported_encryption_names]),
+        OptString.new('KEY', [false, 'The key to use. If not specified, the key will be generated from th password']),
+        OptString.new('PASSWORD', [false, 'The password. If not specified, the KEY option will be used']),
+        OptString.new('SALT', [false, 'The salt to use when creating a key from the password. If not specified, this will be generated from the principal name']),
+        OptInt.new('KVNO', [true, 'The kerberos key version number', 1]),
+        OptEnum.new('OUTPUT_FORMAT', [true, 'The output format to use for listing keytab entries', 'table', %w[csv table]]),
+      ]
+    )
+  end
+
+  def run
+    if datastore['KEYTAB_FILE'].blank?
+      fail_with(Failure::BadConfig, 'Invalid key tab file')
```

Comment:
This message is a little misleading, if you're doing `ADD` it will create a new keytab file, this message implies (to me at least) that it needs to be a valid keytab file already